kernel/common/binding.rb

```diff
@@ -3,8 +3,16 @@ class Binding
   attr_accessor :compiled_code
   attr_accessor :constant_scope
   attr_accessor :proc_environment
-  attr_accessor :self
   attr_accessor :location
+  attr_reader   :receiver
+
+  def self=(obj)
+    @receiver = obj
+  end
+
+  def self
+    @receiver
+  end
```

Comment:
Do we even need to keep `#self` and `#self=` around?

If it's not part of a Ruby-language-compatibility contract, and only Rubinius internal code uses these, can we just make them `attr_accessor :receiver` instead?